apps/marketplace/actions/briefActions.js

```diff
@@ -1,25 +1,17 @@
-import {
-  BRIEF_INFO_FETCH_DATA_SUCCESS,
-  BRIEF_INFO_HAS_ERRORED,
-  BRIEF_RESPONSE_SUCCESS,
-  BRIEF_RESPONSE_FAILURE,
-  BRIEF_RESPONSE_DUPLICATE_FAILURE
-} from '../constants/constants'
+import { BRIEF_INFO_FETCH_DATA_SUCCESS, BRIEF_RESPONSE_SUCCESS } from '../constants/constants'
+import { GENERAL_ERROR } from '../constants/messageConstants'
 
 import dmapi from '../services/apiClient'
-import { sendingRequest } from './appActions'
+import { sendingRequest, setErrorMessage } from './appActions'
 
 export const handleBriefInfoSuccess = response => ({
   type: BRIEF_INFO_FETCH_DATA_SUCCESS,
   brief: response.data
 })
 
-export const handleBriefInfoFailure = response => ({
-  type: BRIEF_INFO_HAS_ERRORED,
-  // eslint-disable-next-line
-  errorMessage: response && response.data ? response.data.message : 'unknown server error'
-})
-
+export const handleBriefInfoFailure = response => dispatch => {
+  dispatch(setErrorMessage(response && response.data && response.data.message ? response.data.message : GENERAL_ERROR))
```

Comment:
don't we control whats in the response?  could you not ensure there's a message in the api?

Reply:
what about a network error (ie. when I forget to start API ;) )

Reply:
we can catch this case in the api client (see my slack msg):

```
const sanitiseErrorResponse = data => {
  if (!data || data.slice(0, 2) === "<?") { // when unhandled api errors return xml
    return GENERAL_ERROR
  } else {
    return data
  }
}

...
.catch(error => {
      if (error.response) {
         return {
             ...
             errorMessage: sanitiseErrorResponse(error.response.data), // data is a string

```
then handleFailure can be:
~~dispatch(handleBriefInfoFailure(response))~~
```
dispatch(setErrorMessage(response.errorMessage))
```